### PR TITLE
Build `scie-pants` using `science`.

### PR DESCRIPTION
**Pull request by jsirois**
Instead of using raw `scie-jump` and a hand-crafted JSON lift manifest,
use `science` and a TOML lift manifest to build `scie-pants`.

The `science` tool and its more friendly interface are the present and
future of end-user scie construction. We convert here and gain a few
things:
+ A manifest format we can comment.
+ Abstraction over platforms and PBS details.
+ Support for embedded build provenanace information for more
  comprehensive black box debugability.
+ A clean path to publishing both "thin" and "fat" binaries, solving
  #154.

### CODE REVIEW

**Comment by jsirois** [comment]
N.B.: I won't land this until science 0.1.2 is available (See: https://github.com/a-scie/lift/pull/21).
With that, the lift manifest build info is like so:
```json
$ SCIE=inspect dist/scie-pants-linux-x86_64 | jq .science
{
  "app_info": {
    "repo": "https://github.com/pantsbuild/scie-pants"
  },
  "binary": {
    "hash": "8b2a9743af6957d49b5b501a50a1a92ccf5aebcd575261127e457b10294814b8",
    "size": 6468367,
    "url": "https://github.com/a-scie/lift/releases/download/v0.1.2/science-linux-x86_64",
    "version": "0.1.2"
  },
  "git_state": "v0.7.2-4-g0943208-dirty",
  "manifest": {
    "hash": "5b79b2d2055380116e8cbf684f230f3af1f816894d16765a8b5bde8ddd19b379",
    "size": 5300,
    "source": "package/scie-pants.toml"
  },
  "notes": [
    "This scie lift JSON manifest was generated from a source lift toml manifest using the science binary.",
    "Find out more here: https://github.com/a-scie/lift/blob/v0.1.2/README.md"
  ]
}
```

And the zip-trailer is preserved:
```console
$ cat <(unzip -l dist/scie-pants-linux-x86_64 | head) <(echo ...) <(unzip -l dist/scie-pants-linux-x86_64 | tail)
warning [dist/scie-pants-linux-x86_64]:  6308880 extra bytes at beginning or within zipfile
  (attempting to process anyway)
Archive:  dist/scie-pants-linux-x86_64
  Length      Date    Time    Name
---------  ---------- -----   ----
        0  1980-01-01 00:00   .bootstrap/
        0  1980-01-01 00:00   .bootstrap/pex/
      401  1980-01-01 00:00   .bootstrap/pex/__init__.py
      235  1980-01-01 00:00   .bootstrap/pex/__main__.py
      601  1980-01-01 00:00   .bootstrap/pex/argparse.py
    11259  1980-01-01 00:00   .bootstrap/pex/atomic_directory.py
      483  1980-01-01 00:00   .bootstrap/pex/attrs.py
warning [dist/scie-pants-linux-x86_64]:  6308880 extra bytes at beginning or within zipfile
  (attempting to process anyway)
...
     4897  1980-01-01 00:00   scie_pants/install_pants.py
     1680  1980-01-01 00:00   scie_pants/log.py
    43256  1980-01-01 00:00   scie_pants/pants_release_tags.json
     6665  1980-01-01 00:00   scie_pants/pants_version.py
     1601  1980-01-01 00:00   scie_pants/ptex.py
     1474  1980-01-01 00:00   scie_pants/record_scie_pants_info.py
      346  1980-01-01 00:00   scie_pants/scie_pants.iml
     9605  1980-01-01 00:00   scie_pants/update_scie_pants.py
---------                     -------
  4047731                     386 files
```

File: package/scie-pants.toml [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+version = "0.11.0"
+
+[[lift.interpreters]]
+id = "cpython38"
+provider = "PythonBuildStandalone"
+release = "20230507"
+lazy = true
+version = "3.8.16"
+
+[[lift.interpreters]]
```
Comment (jsirois) [inline package/scie-pants.toml:24]: N.B.: Adding 3.11 support just requires adding a similar table with "3.11.3" as the version, an id of "cpython311" and an edit to `tools/src/scie_pants/configure_pants.py` that selects "cpython311" for the appropriate Pants version cutoff.

**Comment by jsirois** [comment]
Alright, some more work is needed here for the Linux CI case which uses a volume mount into an alpine linux container: there are now paths not mounted. There is also investigation needed to find the root of the Mac OS x86_64 issue.

**Comment by jsirois** [comment]
@sureshjoshi yeah, your mac mini error is in Pants, not scie-pants. It's trying to run Pants from sources and getting an error in the `build-support/pants_venv` script.

**Comment by jsirois** [comment]
Aha, OK - the Linux arm issue is specifically the issue that PBS for Linux + arm + musl does not exist as a thing, just Linux + arm + glibc and Linux + x86_64 + {glibc,musl} are things. The final scie build will need to be back on the other side of the alpine image build of the scie-pants Rust binary.

File: package/src/main.rs [hunk 2]
```diff
@@ -204,45 +211,102 @@ fn maybe_build(args: &Args, build_context: &BuildContext) -> Result<Option<PathB
             check,
             tools_pex_mismatch_warn,
         } => {
```
Comment (stuhood) [inline package/src/main.rs:213]: It seems like the four branches of `Commands::Test` (and `Commands::Scie`) could be unified into one branch which initialized the `tools_pex` or `scie_pants` if they weren't set? I know this was here before, but as each branch gets longer (in very similar ways) the (apparent?) duplication is more obvious.

**Review by stuhood — APPROVED** [review]

File: package/src/main.rs (hunk 2 above)
Comment (jsirois) [reply package/src/main.rs:213]: Done here: #194

**Pull request merged**